docs/CONFIGURATION.md

```diff
@@ -0,0 +1,188 @@
+# Configuring the Mountpoint for Amazon S3 CSI Driver
+
+See [the Mountpoint documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md) for
+Mountpoint specific configuration.
+
+## AWS Credentials
+
+The Mountpoint CSI Driver can be configured to ingest credentials via two approaches: globally for the entire 
+Kubernetes cluster, or using credentials assigned to pods.
+
+### Driver Level Credentials
+
+By setting driver-level credentials, the whole cluster uses the same set of credentials. 
+
+Using this configuration, the credentials that are used are set at installation time, either using Service
+Accounts, or using K8s secrets. 
+
+See [install.md](./install.md) for more details on this approach.
+
+**TODO** - Migrate driver level setup here.
+
+### Driver Level Credentials with IRSA
+
+Configuring [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (IRSA)
+is the recommended way to set up the CSI Driver if you want to use Driver Level credentials.
+
+```mermaid
```

Comment:
It would be nice to set a different color on credential source in the diagram if possible. i.e., "IAM Credentials - Driver" would be in a different color here to make it easy to understand